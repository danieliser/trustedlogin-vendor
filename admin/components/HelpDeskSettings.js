import { Form, Submit, Input, FormTable } from "./index";
import { __ } from "@wordpress/i18n";
import { useMemo } from "react";

/**
 * TrustedLogin Settings Form
 */
export default function (props) {
  const { team } = props;

  const helpdesk = useMemo(() => {
    return Array.isArray(team.helpdesk) ? team.helpdesk[0] : team.helpdesk;
  }, [team]);
  const noop = () => {};
  const secret = useMemo(() => {
    if (
      helpdesk &&
      team.helpdesk_settings &&
      team.helpdesk_settings.hasOwnProperty(helpdesk)
    ) {
      return team.helpdesk_settings[helpdesk].secret;
    }
    return "";
  }, [team, helpdesk]);

  const callback = useMemo(() => {
    if (
      helpdesk &&
      team.helpdesk_settings &&
      team.helpdesk_settings.hasOwnProperty(helpdesk)
    ) {
      return team.helpdesk_settings[helpdesk].callback;
    }
    return "";
  }, [team, helpdesk]);
  return (
    <>
      <Input
        name="helpscout[secret]"
        label={__("Help Scout Secret Key")}
        value={secret}
        onChange={noop}
        disabled={true}
      />
      <Input
        name="helpscout[callback]"
        label={__("Help Scout Callback URL	")}
        value={callback}
        type="url"
        disabled={true}
        onChange={noop}
      />
    </>
  );
}
