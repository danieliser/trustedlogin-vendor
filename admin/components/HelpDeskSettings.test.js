import { render, fireEvent, cleanup, act } from "@testing-library/react";
import HelpDeskSettings from "./HelpDeskSettings";

describe("HelpDeskSettings", () => {
	afterEach(cleanup);
	let settings = {
		isConnected: false,
		teams: [
			{
				id: 0,
				account_id: "7",
				private_key: "pk_1",
				api_key: "ak_1",
				helpdesk: "helpscout",
				approved_roles: ["editor"],
			},
		],
		helpscout: {
			callback: "",
			secret: "",
		},
	};
	it("renders with save enabled", () => {
		const onSave = jest.fn();

		const { container } = render(
			<HelpDeskSettings settings={settings} canSave={false} onSave={onSave} />
		);
		expect(container).toMatchSnapshot();
	});

	it("updates helpscout settings", () => {
		const onSave = jest.fn();
		const setSettings = jest.fn();
		const { getByLabelText } = render(
			<HelpDeskSettings
				onSave={onSave}
				settings={settings}
				canSave={true}
				setSettings={setSettings}
			/>
		);
		//Change secret
		act(() => {
			fireEvent.change(getByLabelText("Help Scout Secret Key"), {
				target: { value: "42" },
			});
		});

		expect(setSettings).toBeCalledWith({
			...settings,
			helpscout: {
				...settings.helpscout,
				secret: "42",
			},
		});

		//Change callback
		act(() => {
			fireEvent.change(getByLabelText("Help Scout Callback URL"), {
				target: { value: "https://happy.dog" },
			});
		});

		expect(setSettings).toBeCalledWith({
			...settings,
			helpscout: {
				...settings.helpscout,
				callback: "https://happy.dog",
			},
		});
	});
});
