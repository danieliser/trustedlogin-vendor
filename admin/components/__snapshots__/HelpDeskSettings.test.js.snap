// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDeskSettings renders with save enabled 1`] = `
<div>
  <form>
    <table
      class="form-table"
      role="presentation"
    >
      <tbody>
        <tr>
          <th
            scope="row"
          >
            <label
              for="helpscout[secret]"
            >
              Help Scout Secret Key
            </label>
          </th>
          <td>
            <input
              class="regular-text ltr"
              id="helpscout[secret]"
              name="helpscout[secret]"
              type="text"
              value=""
            />
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <label
              for="helpscout[callback]"
            >
              Help Scout Callback URL	
            </label>
          </th>
          <td>
            <input
              class="regular-text ltr"
              id="helpscout[callback]"
              name="helpscout[callback]"
              type="url"
              value=""
            />
          </td>
        </tr>
      </tbody>
    </table>
    <p
      class="submit"
    >
      <input
        class="button button-secondary button-hero"
        disabled=""
        type="submit"
        value="Save"
      />
    </p>
  </form>
</div>
`;
